test/logic/gossip_node_logic_test.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class GossipNodeLogicTest < ActiveSupport::TestCase
+  include GossipNodeLogic
+
+  setup do
+    @mainnet_url = "https://api.mainnet-beta.solana.com"
+    @network = "mainnet"
+
+    @payload = { network: @network, config_urls: [@mainnet_url] }
+
+    @json_data = File.read("#{Rails.root}/test/json/gossip_nodes.json")
+  end
+
+  test "get_nodes adds gossip nodes to pipeline" do
+    SolanaCliService.stub(:request, @json_data, ["gossip", @mainnet_url]) do
+      p = Pipeline.new(200, @payload)
+                  .then(&get_nodes)
+
+      assert_equal 200, p.code
+      refute p.payload[:current_nodes].blank?
```

Comment:
I'd check also some record or count instead of existence, because it might happen that it not blank due to having ie. error message or sth like that.

Reply:
@maciejkocylapc